feat: Review 기능 구현

TaeHee00 opened the pull request.

TaeHee00 committed:
feat: create temp Course Entity

Course Review 구현을 위한 임시 Course 엔티티 생성

TaeHee00 committed:
Merge remote-tracking branch 'origin/main' into feature/course

TaeHee00 committed:
feat: create EvaluationResultRequest DTO

TaeHee00 committed:
feat: create ReviewCreate DTO

TaeHee00 committed:
feat: create ReviewResponse DTO

TaeHee00 committed:
feat: create TargetType enum class

TaeHee00 committed:
feat: evaluation 관련 Repository 구현

TaeHee00 committed:
feat: Review Repository 구현

TaeHee00 committed:
feat: Course Repository 구현

TaeHee00 committed:
fix: Evaluation field Type

Evaluation - evaluationType class 변경
기존 Enum Type -> String

* 변경사유: DB에서 컬럼으로 저장되어 있어 만약 다른 type 추가 시 DB insert, Enum Type 추가 필요 -> 두번 작업해야함

TaeHee00 committed:
remove: Evaluation Type 삭제

Evaluation Type 미사용으로 인한 삭제처리

TaeHee00 committed:
refactor: Course Entity Table 이름 오타 수정 및 import 정리

TaeHee00 committed:
feat: Create Review Result Entity

TaeHee00 committed:
feat: Evaluation Service

Course Entity 조회 및 존재 여부 확인 기능 구현

TaeHee00 committed:
feat: Evaluation Service 구현

평가 항목 등록, 평가 항목 질문 등록, 평가 항목 답변 등록, 엔티티 조회, 엔티티 존재여부 확인 등 관련 메서드 구현

TaeHee00 committed:
doc: ErrorMessage 등록

TaeHee00 committed:
feat: Repository Query Method 추가

TaeHee00 committed:
remove: 사용하지 않는 DTO 삭제

TaeHee00 committed:
feat: 요청 관련 DTO 추가

TaeHee00 committed:
feat: 반환 관련 DTO 추가

TaeHee00 committed:
fix: (ReviewCreateRequest) file 대신 FileMaster ID를 받도록 DTO 변경

TaeHee00 committed:
chore: ErrorCode 추가

TaeHee00 committed:
feat: Review Entity update 메서드 추가

TaeHee00 committed:
fix: EvaluationResult Entity 답변 변경이 가능하도록 val -> var 변경

TaeHee00 committed:
feat: Review Response DTO가 평가 결과도 반환하도록 변경

TaeHee00 committed:
feat: Review Service 구현

TaeHee00 committed:
feat: FileMaster 조회 기능 구현

TaeHee00 committed:
feat: Doc수정 및 조회, 체크 기능 구현

TaeHee00 committed:
feat: Review Controller 구현

TaeHee00 committed:
feat: TestCode

TaeHee00 committed:
Merge remote-tracking branch 'origin/feature/course' into feature/course

TaeHee00 committed:
fix: CustomValidator 적용

TaeHee00 committed:
fix: JPA -> QueryDSL 변경

TODO: Valid 관련 오류로 DTO Valid 주석처리, CustomValidator Merge 이후 주석 해제 예정,

TaeHee00 committed:
Merge pull request #46 from TaeHee00/feature/course

Feature/course Review CURD

TaeHee00 committed:
Merge branch 'main' into feature/course

TaeHee00 committed:
fix: change ErrorCode.kt convention

messages 확장자 변경에 따른 message code 컨벤션 변경

TaeHee00 committed:
fix: change messages properties -> messages yml

기존 메시지 형식 변경

TaeHee00 committed:
delete: 머지로 인해 생긴 중복 함수 삭제

TaeHee00 committed:
fix: import 변경

TaeHee00 committed:
fix: change messages properties -> messages yml

TaeHee00 committed:
fix: change messages properties -> messages yml

TaeHee00 committed:
fix: 파일 위치 변경

TaeHee00 committed:
feat: add Evaluation Repository find by ID

TaeHee00 committed:
refactor: EvaluationCustomRepository for flexible querying

TaeHee00 committed:
fix: 사용하지 않는 컬럼 삭제 및 Evaluation find 메서드 mocking 변경

TaeHee00 committed:
feat: role-mapping api 추가

/api/reviews

TaeHee00 committed:
fix: OAuth 추가로 인한 RequestDTO userId field 분리

TaeHee00 committed:
fix: RequestDTO 변경으로 인한 Test 변경

TaeHee00 committed:
chore: ErrorCode 추가

TaeHee00 committed:
bugfix: find 쿼리에서 from 절이 없어 오류 발생

추가 후 테스트 완료

TaeHee00 committed:
feat: 권한관련 Error 추가

TaeHee00 committed:
feat: 권한 확인용 메서드 추천

TaeHee00 committed:
fix: DTO에서 userId 필드 삭제

-> Token Interceptor 사용하여 request 인자로 추가로 받음

TaeHee00 committed:
feat: 요청 인자 변경

TaeHee00 committed:
fix: Validator 적용

TaeHee00 committed:
fix: DTO 및 Validator 재적용

EnumValidator 적용을 위해 DTO field 중 Enum field String type으로 변경, Service Layer 또한 valueOf을 사용하여 원래 Enum Type으로 변경

TaeHee00 committed:
fix: remove deleteReview request Dto

deleteReview api 요청 시 DTO 대신 requestParam으로 받는 형식으로 변경
* userId 분리로 인하여 DTO 내 field 1개만 남아  requestParam이 효율적이라 생각되어 변경

TaeHee00 committed:
fix: review 수정, 답변 수정 로직 수적

Admin이 수정했을 경우 답변 작성 user 이름 등록시 Admin ID로 입력되어 무결성 오류,
ex: reviewId는 유저이나 그와 연결된 answer는 Admin이 등록된 형식

TaeHee00 committed:
fix: review delete api

requestParam 대신 pathVariable을 사용해 restfull하게 변경

TaeHee00 committed:
fix: review api 와일드카드 변경

TaeHee00 committed:
feat: Review 엔티티 SQLDelete 추가

TaeHee00 committed:
feat: Review Create parameter 변경

TaeHee00 committed:
feat: EvaluationResult Entity SQLDelete 추가

TaeHee00 committed:
fix: EvaluationResult Repository update 메서드 변경

QueryDSL 사용시 SQLDelete가 적용이 안되어 업데이트로 직접 변경

TaeHee00 committed:
refactor: ktlint 적용

TaeHee00 committed:
fix: TargetType 관련 수정

TaeHee00 committed:
Merge pull request #55 from TaeHee00/feature/course

Feature/course

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/controller/ReviewController.kt:
```diff
@@ -0,0 +1,102 @@
+    fun deleteReview(
+        @PathVariable reviewId: UUID,
+        @AuthenticationPrincipal user: CustomOAuth2User,
+    ): ResponseEntity<Unit> {
```
Unit이 Void역할인줄 첨알았네요 조아용

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/GetReviewListRequest.kt:
```diff
@@ -0,0 +1,18 @@
+package com.dugaza.letsdrive.dto.review
+
+import com.dugaza.letsdrive.service.TargetType
+import com.dugaza.letsdrive.validator.CustomValidator
+import com.dugaza.letsdrive.validator.ValidEnum
+import java.util.UUID
+
+class GetReviewListRequest(
+    @field:CustomValidator.NotNull(message = "targetId는 필수 입력값입니다.")
+    val targetId: UUID,
+    @field:ValidEnum(
+        enumClass = TargetType::class,
+        message = "type은 COURSE 만 가능합니다.",
+        allowNull = false,
+        ignoreCase = false,
+    )
+    val targetType: String,
+)
```
클라이언트에게 받는 request body는 vo입니다 파일 이동해주세영

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/ModifyReviewRequest.kt:
```diff
@@ -0,0 +1,20 @@
+package com.dugaza.letsdrive.dto.review
+
+import com.dugaza.letsdrive.validator.CustomValidator
+import java.util.UUID
+
+class ModifyReviewRequest(
+    @field:CustomValidator.NotNull(message = "reviewId는 필수 입력값입니다.")
+    val reviewId: UUID,
+    @field:CustomValidator.NotNull(message = "evaluationResultList는 필수 입력값입니다.")
+    val evaluationResultList: List<UUID>,
+    @field:CustomValidator.NotNull(message = "score는 필수 입력값입니다.")
+    val score: Double,
+    @field:CustomValidator.NotBlank(message = "content는 필수 입력값입니다.")
+    @field:CustomValidator.Size(
+        min = 1,
+        max = 4000,
+        message = "content는 1자 이상 4000자 이하로 작성해주세요.",
+    )
+    val content: String,
+)
```
클라이언트에게 받는 request body는 vo입니다 파일 이동해주세영

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/GetReviewListRequest.kt:
```diff
@@ -0,0 +1,18 @@
+    @field:ValidEnum(
+        enumClass = TargetType::class,
+        message = "type은 COURSE 만 가능합니다.",
+        allowNull = false,
+        ignoreCase = false,
```
default로 false설정 해놔서 이 부분은 지워도 false로 적용돼요

TaeHee00 replied:
GetReviewListReques 요청 DTO 삭제 이후 RequestParam으로 변경되어 DTO 삭제했습니다!

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/ReviewCreateRequest.kt:
```diff
@@ -0,0 +1,33 @@
+package com.dugaza.letsdrive.dto.review
+
+import com.dugaza.letsdrive.service.TargetType
+import com.dugaza.letsdrive.validator.CustomValidator
+import com.dugaza.letsdrive.validator.ValidEnum
+import java.util.UUID
+
+class ReviewCreateRequest(
+    @field:CustomValidator.NotNull(message = "targetId는 필수 입력값입니다.")
```
클라이언트에게 받는 request body는 vo입니다 파일 이동해주세영

TaeHee00 replied:
클라이언트 요청으로 받는값은 DTO로 알고있었는데 VO로 하는게 맞나요..?

roy0424 replied:
VO로 하기로 했었어여 제가 하나 만들어놓은거 있어요
vo.user.ChangeEmail

vo는 불변성을 가지고 있어서 request를 받기 좋아요

TaeHee00 replied:
저도 같은 생각입니당 vo 쓰는곳에서는 vo만 쓰고 dto 쓰는곳에서는 dto만 쓰던거같아서
휴가 복귀 하시면 한번 정하시죠! 이거 dto -> vo 변경건만 빼고 나머지만 resolve 해두겠습니당

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/ReviewCreateRequest.kt:
```diff
@@ -0,0 +1,33 @@
+    @field:ValidEnum(
+        enumClass = TargetType::class,
+        message = "Types are only available for Course.",
+        allowNull = false,
+        ignoreCase = false,
```
default로 false설정 해놔서 이 부분은 지워도 false로 적용돼요

TaeHee00 replied:
수정완료했습니다!

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/ReviewCreateRequest.kt:
```diff
@@ -0,0 +1,33 @@
+    val evaluationId: UUID,
+    @field:ValidEnum(
+        enumClass = TargetType::class,
+        message = "Types are only available for Course.",
```
이 친구만 영어로 메시지가 되어있어요

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/entity/common/evaluation/Evaluation.kt:
```diff
@@ -3,14 +3,11 @@ package com.dugaza.letsdrive.entity.common.evaluation
-    @Enumerated(EnumType.STRING)
     @Column(name = "evaluation_type", nullable = false)
-    val type: EvaluationType,
+    val type: String,
```
이 부분은 enum으로 해야할거같아요

TaeHee00 replied:
요거 엔티티 자체에 type값만 있어서 Enum으로 만들면 DB에 값 넣을때마다 Enum 값 추가해줘야해서 String으로 두는게 좋을꺼같아서 이렇게 만들었어요

roy0424 replied:
타입이 정해진 값이 아닌건가요 ?

TaeHee00 replied:
EvaluationType으로 관리하면 확실히 개발할때 휴먼에러도 줄어들고 좋을꺼같은데
Evaluation에 현재 실질적으로 사용하는 컬럼이 evaluation_type 이거 하나라서 Evaluation 엔티티가 Evaluation Type의 역할을 하고있고 Evaluation 테이블에서 저장하게 될 데이터를 넣을때 마다 Enum에 추가해줘야하는 상황이 발생해서 String으로 관리하는게 이후 개발에 편할꺼같아요

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/entity/course/Course.kt:
```diff
@@ -0,0 +1,22 @@
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id", nullable = false, foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
+    val user: User,
+    @Column(name = "course_name", nullable = false)
```
칼럼이름을 따로 지정해줄 필요없어보이네요

TaeHee00 replied:
<img width="539" alt="스크린샷 2025-03-07 22 55 18" src="https://github.com/user-attachments/assets/f98daf4f-cf9f-4b3b-8911-99e2eda73353" />
DB 테이블에 이름이 course_name이라 name 빼고 실행해보니까 컬럼을 못찾네요... ㅠ

roy0424 replied:
DB 칼럼 수정해주세여 !

TaeHee00 replied:
변경 완료했습니둥!

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/course/CourseRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository
+import java.util.UUID
+
+@Repository
```
https://velog.io/@novo2003/JpaRepository-%EC%99%80-Bean-%EB%93%B1%EB%A1%9D
spring에서는 repository를 자동으로 인식해주기때문에 해당 어노테이션을 안달아도 돼여

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationAnswerCustomRepositoryImpl.kt:
```diff
@@ -0,0 +1,27 @@
+                .selectOne()
+                .from(evaluationAnswer)
+                .where(
+                    evaluationAnswer.question.eq(question)
```
실제 쿼리가 어떻게 날아가는지 확인 필요할거같아요
쿼리가 id로 where하는거면 상관없지만 각 속성을 다 비교한다면 id로 where문 작성해야할거같아요

TaeHee00 replied:
QueryDSL 관련 CustomRepository 전부 테스트 코드 작성 완료했습니다!

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationAnswerRepository.kt:
```diff
@@ -0,0 +1,15 @@
+import org.springframework.stereotype.Repository
+import java.util.UUID
+
+@Repository
```
여기도 마찬가지로 어노테이션 필요없어용

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationCustomRepository.kt:
```diff
@@ -0,0 +1,13 @@
+
+interface EvaluationCustomRepository {
+    fun find(
+        id: UUID? = null,
+        type: String? = null,
```
두값 다 조회시 nullable 하지 않은거같아보여요
nullable하지 않게 해주세용

TaeHee00 replied:
조건부로 인자를 받아서 들어온 값만 조건으로 검색하도록 만들어둔 거라 
일부로 유연하게 만든 find 함수에요!

혹시 인자가 하나도 안들어올 경우를 대비해서 아래같이 check 메서드 따로 추가했습니다.

<img width="495" alt="스크린샷 2025-03-08 18 41 41" src="https://github.com/user-attachments/assets/32680f09-492e-48d4-ae19-18b940cbaf95" />
<img width="397" alt="스크린샷 2025-03-08 18 41 53" src="https://github.com/user-attachments/assets/320519a4-032b-4312-8a55-6fdac42bb228" />

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationQuestionCustomRepositoryImpl.kt:
```diff
@@ -0,0 +1,39 @@
+                .selectOne()
+                .from(evaluationQuestion)
+                .where(
+                    evaluationQuestion.evaluation.eq(evaluation)
```
이 부분도 쿼리보고 판단해서 id만 검사하는지 확인 해야할거같아여

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationQuestionRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository
+import java.util.UUID
+
+@Repository
```
이 부분도 어노테이션 제거해주세여

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository
+import java.util.UUID
+
+@Repository
```
제거해주세여ㅑ

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationResultCustomRepositoryImpl.kt:
```diff
@@ -0,0 +1,70 @@
+    override fun findAll(
+        userId: UUID,
+        reviewId: UUID,
+    ): List<EvaluationResult> {
```
pagination이 필요할거같아여

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/evaluation/EvaluationResultRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository
+import java.util.UUID
+
+@Repository
```
제거해주세영

TaeHee00 replied:
전부 삭제 했습니다!

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/repository/review/ReviewRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import java.util.UUID
+
+interface ReviewRepository : JpaRepository<Review, UUID> {
+    fun findAllByTargetId(targetId: UUID): List<Review>
```
paging해주세영

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/service/course/CourseService.kt:
```diff
@@ -0,0 +1,38 @@
+     * @exception BusinessException ErrorCode.COURSE_NOT_FOUND
+     */
+    fun existsCourseById(courseId: UUID) {
+        val result = courseRepository.existsById(courseId)
+        if (!result) throw BusinessException(ErrorCode.COURSE_NOT_FOUND)
```
이 부분 ?: 로 처리하면 깔끔할거같아요

TaeHee00 replied:
JPA 디폴드 메서드 이용한거라 QueryDSL로 메서드 합치고 테스트 이후에 변경하겠습니다!

TaeHee00 replied:
<img width="526" alt="스크린샷 2025-03-09 22 19 42" src="https://github.com/user-attachments/assets/91fcf150-08ea-41f2-8d49-6ff319c69808" />

만들면서 생각해보니까 exists 메서드라 반환값이 boolean 타입으로 되어있어서
takeIf 사용해서 처리했습니다!

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/service/review/ReviewService.kt:
```diff
@@ -0,0 +1,398 @@
+     */
+    @Transactional
+    fun createReview(
+        request: ReviewCreateRequest,
```
service layer에서는 다른 컨틀롤러 및 서비스에서 쓰일 가능성이 있어서 확장성을 위해 인자는 하나하나 다 받아주는게 좋을거같아요

TaeHee00 replied:
<img width="450" alt="스크린샷 2025-03-07 23 21 06" src="https://github.com/user-attachments/assets/9e041396-d0f6-464c-bdd2-4fc20ccdab8e" />

이거 인자 풀다가 생각난건데 fileMaster 생성조건이 궁금해졌어요!
무조껀 review같은거 만들면 생기는지 업로드 시 생기는지!
업로드 할때 만든다고 했던거 같은데 확실하게 기억이 안나서... ㅠ
코멘트나 톡 남겨주시면 추가 수정 하겠습니다!

roy0424 replied:
FileMaster 는 유저가 리뷰 작성시 사진추가하기 같은 버튼을 누르고 추가했을 경우 생기게 돼요 그래서 널러블하다고 보면 좋을거같아요

roy0424 commented on src/main/kotlin/com/dugaza/letsdrive/service/review/ReviewService.kt:
```diff
@@ -0,0 +1,398 @@
+            )
+
+        request.evaluationResultList
+            .stream()
```
stream은 안해도 될거같아요

TaeHee00 replied:
오... 감사합니다... 이런 꿀팁이...

TaeHee00 committed:
fix: review 등록 함수명 변경

컨벤션 따라서 동사로 변경

TaeHee00 committed:
fix: Review Controller 인자 줄바꿈 적용

TaeHee00 committed:
fix: review 조회 API 변경

TaeHee00 committed:
refactor: review 조회용 DTO 삭제

TaeHee00 committed:
fix: review DTO 삭제 이후 Test 코드 변경

TaeHee00 commented on src/main/kotlin/com/dugaza/letsdrive/controller/ReviewController.kt:
```diff
@@ -0,0 +1,102 @@
+
+    @GetMapping
+    fun getReviewList(
+        @RequestBody @Valid request: GetReviewListRequest,
```
<img width="844" alt="스크린샷 2025-03-07 22 40 00" src="https://github.com/user-attachments/assets/3faab675-0a36-430a-a046-c2f828d49f2a" />
변경 완료했습니다!

TaeHee00 committed:
fix: 명시적으로 선언한 annotation default option 삭제

TaeHee00 committed:
fix: review create dto throw message 변경

일관성있게 한국어로 변경

TaeHee00 committed:
refactor: @Repository annoation delete

TaeHee00 committed:
refactor: review create 메서드 인자 변경

이후 다른 service 에서 사용할 경우 dto를 사용하면 함수 이용이 어려워 인자에 dto 컬럼 분해하여 적용

TaeHee00 committed:
feat: QueryDSL 조건 검사를 위한 Object class 추가

TaeHee00 committed:
feat: Evaluation answer repository exists 메서드 인자 수정

TaeHee00 committed:
fix: Checks 메서드를 이용하도록 변경

TaeHee00 committed:
feat: EvaluationAnswerCustomRepositoryImpl Test

QueryDSL 테스트 전용 코드 구현

TaeHee00 committed:
fix: display name 변경

TaeHee00 committed:
feat: EvaluationCustomRepositoryImpl Test

QueryDsl 전용 테스트 코드

TaeHee00 committed:
fix: display name change

TaeHee00 committed:
feat: EvaluationQuestionCustomRepositoryImpl Test

QueryDSL 메서드 관련 테스트 코드 추가

TaeHee00 committed:
feat: evaluation result delete 에러 핸들링

revieId로 삭제할 경우 해당하는 evaluation result list가 없을 경우 에러 발생 (존재하지 않는 review ID 혹은 잘못된 평가)

TaeHee00 committed:
feat: EvaluationResultCustomRepositoryImpl Test

QueryDSL 관련 메서드 테스트 코드 작성

TaeHee00 committed:
fix: Course Entity name column name change

TaeHee00 committed:
feat: courseRepository QueryDSL로 마이그레이션

TaeHee00 committed:
check point

downingmoon commented on src/main/kotlin/com/dugaza/letsdrive/dto/review/GetReviewListRequest.kt:
```diff
@@ -0,0 +1,18 @@
+import java.util.UUID
+
+class GetReviewListRequest(
+    @field:CustomValidator.NotNull(message = "targetId는 필수 입력값입니다.")
```
얘는 코드로 바꾸지 않았었나...요?

downingmoon replied:
아 예전 버전이구나 나중에 합치고 점검한번 해봐야 할듯

TaeHee00 committed:
feat: Review Find Query 성능 개선

기존 단순 쿼리를 여러번 보내 조회하는 쿼리에서 한방쿼리로 변경함, 요청 시간 3~5sec에서 1sec 이하로 성능 개선

TaeHee00 committed:
fix: Review find query 변경으로 인한 Interface 파일 수정

TaeHee00 committed:
fix: Review 조회 시 작성 User의 데이터를 같이 조회

TaeHee00 committed:
refactor: 사용하지 않는 import 삭제

TaeHee00 committed:
fix: 쿼리 파라미터 변경으로 인한 수정

TaeHee00 committed:
feat: review find api 반환 타입 변경

TaeHee00 committed:
fix: registerReview vo로 변경

TaeHee00 committed:
fix: vo 이름 변경

TaeHee00 committed:
fix: vo 이름 변경

TaeHee00 committed:
fix: QueryDSL 사용으로 인한 테스트 코드 변경

TaeHee00 committed:
fix: change convention

TaeHee00 committed:
fix: ktlint

TaeHee00 commented on src/main/kotlin/com/dugaza/letsdrive/service/TargetType.kt:
```diff
@@ -0,0 +1,5 @@
+package com.dugaza.letsdrive.service
+
+enum class TargetType {
+    COURSE,
```
이슈 등록해두고 이후 테스크에서 진행하겠습니다!
TargetType 다국어 지원 #58

TaeHee00 committed:
fix: review response of method

TaeHee00 committed:
Merge pull request #59 from TaeHee00/feature/course

review 코멘트 반영

TaeHee00 committed:
fix: Evaluation - EvaluationType 지정

TaeHee00 committed:
Merge pull request #65 from TaeHee00/feature/course

fix: Evaluation - EvaluationType 지정
